src/OpenTelemetry.Extensions/Trace/PartialActivityProcessor.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Extensions.Trace;
+
+/// <summary>
+/// Activity processor that exports LogRecord on Activity start, during it's lifetime as a heartbeat and on end.
+/// </summary>
+/// <remarks>
+/// Add this processor *before* exporter related Activity processors.
+/// </remarks>
+public class PartialActivityProcessor : BaseProcessor<Activity>
+{
+    private readonly ManualResetEvent shutdownTrigger;
+    private readonly BaseExporter<LogRecord> logExporter;
+    private readonly BaseProcessor<LogRecord> logProcessor;
+    private readonly ILogger<PartialActivityProcessor> logger;
+    private ILoggerFactory loggerFactory;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PartialActivityProcessor"/> class.
+    /// </summary>
+    /// <param name="logExporter">Log exporter to be used.</param>
+    public PartialActivityProcessor(BaseExporter<LogRecord> logExporter)
+    {
+#if NET
+        ArgumentNullException.ThrowIfNull(logExporter);
+#else
+        if (logExporter == null)
+        {
+            throw new ArgumentOutOfRangeException(nameof(logExporter));
+        }
+#endif
+        this.logExporter = logExporter;
+        this.logProcessor = new SimpleLogRecordExportProcessor(logExporter);
+
+        // Configure OpenTelemetry logging to use the provided logExporter
+        this.loggerFactory = LoggerFactory.Create(builder =>
+        {
+            builder.ClearProviders();
+            builder.AddOpenTelemetry(options =>
+            {
+                options.IncludeScopes = true;
+                options.AddProcessor(this.logProcessor);
+            });
+        });
+        this.logger = this.loggerFactory.CreateLogger<PartialActivityProcessor>();
+
+        this.shutdownTrigger = new ManualResetEvent(false);
+    }
+
+    /// <inheritdoc />
+    public override void OnStart(Activity data)
+    {
+        if (data == null)
+        {
+            return;
+        }
+
+        using (this.logger.BeginScope(GetStartLogRecordAttributes()))
+        {
+            this.logger.LogInformation(
+                ActivitySpec.Json(new ActivitySpec(data, ActivitySpec.Signal.Heartbeat)));
+        }
+    }
+
+    /// <inheritdoc />
+    public override void OnEnd(Activity data)
```

Comment:
This means we'd send a heartbeat on _every_ activity. This will result in a lot of heartbeat logs for regular actions that don't take a long time. Also, it means a long running action that doesn't continually create activities won't generate heartbeat messages.

Reply:
i'm going to introduce heartbeat dealy which is intended for those regular actions that don't take long time. nice observation!